keps/sig-network/1672-tracking-terminating-endpoints/README.md

```diff
@@ -148,9 +155,123 @@ of the [EndpointSlice API work](/keps/sig-network/20190603-endpointslices/README
 Since this is an addition to the EndpointSlice API, the version skew strategy will follow that
 of the [EndpointSlice API work](/keps/sig-network/20190603-endpointslices/README.md).
 
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: EndpointSliceTerminatingCondition
+  - Components depending on the feature gate: kube-apiserver and kube-controller-manager
+
+###### Does enabling the feature change any default behavior?
+
+Yes, terminating endpoints are now included as part of EndpointSlice API. The `ready` condition of an EndpointSlice will always be `false` to ensure consumers do not send traffic to terminating endpoints unless new conditions are checked.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. On rollback, terminating endpoints will no longer be included in EndpointSlice and the `terminating` and `serving` conditions will not be set.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+EndpointSlice will continue to have the `terminating` and `serving` condition set.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes, there will be integration and e2e tests validating whether EndpointSlice contains endpoints for pods that are terminating.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout fail? Can it impact already running workloads?
+
+If there are consumers of EndpointSlice that do not check the `ready` condition, then they may unexpectedly start sending traffic to terminating endpoints.
+It is assumed that almost all consumers of EndpointSlice check the `ready` condition prior to allowing traffic to a pod.
+
```

Comment:
One can also imagine bugs in endpointslice controller (or strategy itself) leading to crashes or individual endpoints marked incorrectly...

Reply:
Sure, but is that really a problem for this specific issue or a generic EndpointSlice / controller problem?

Reply:
But you're changing endpointslice controller as part of this KEP.

Reply:
Sure, but endpointslice controller does not expose metrics for existing conditions or other parameters. Are you suggesting we only add metrics for terminating condition or for everything else in the EndpointSlice API?

Reply:
This feels like a dup of https://github.com/kubernetes/enhancements/pull/2725#discussion_r631599425 - let's discuss there?